api/Trace/Events.php

```diff
@@ -15,6 +15,15 @@ interface Events extends \IteratorAggregate, \Countable
      */
     public function addEvent(string $name, ?Attributes $attributes = null, ?int $timestamp = null): Events;
 
+    /**
+ * Adding an exception event should not invalidate nor change any existing iterators.
+     * @param string $name
+     * @param Attributes|null $attributes
+     * @param int|null $timestamp
+     * @return Events Must return $this to allow setting multiple attributes at once in a chain.
+     */
+    public function recordException(string $name, ?Attributes $attributes = null, ?int $timestamp = null): Events;
```

Comment:
For the `Events` collection should not be difference between a regular event and exception (see comment for the `Span` interface). So the method `Events::recordException()` may be redundant.

Reply:
Makes sense @lalex , without an exception object, recordException is just another alias for addEvent, I will look into the links and update the PR, thanks.